x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/esql.ts

```diff
@@ -70,38 +80,56 @@ export const esqlExecutor = async ({
   } = sharedParams;
   const loggedRequests: RulePreviewLoggedRequest[] = [];
   const ruleParams = completeRule.ruleParams;
-  /**
-   * ES|QL returns results as a single page. max size of 10,000
-   * while we try increase size of the request to catch all alerts that might been deduplicated
-   * we don't want to overload ES/Kibana with large responses
-   */
-  const ESQL_PAGE_SIZE_CIRCUIT_BREAKER = tuple.maxSignals * 3;
   const isLoggedRequestsEnabled = state?.isLoggedRequestsEnabled ?? false;
 
   return withSecuritySpan('esqlExecutor', async () => {
     const result = createSearchAfterReturnType();
-    let size = tuple.maxSignals;
```

Comment:
I think it would be more clear if we keep `const size = tuple.maxSignals + 1` here and don't add 1 to `size` inside `buildEsqlSearchRequest`, then we can also use `size` on line 277 (https://github.com/elastic/kibana/pull/217069/files#diff-9e5aa7725f437f26beb22fd30fe20ef4601d2e827dc76ccb80d703e8a694f250R277).